bindings/ocaml/Hacl.ml

```diff
@@ -341,3 +299,51 @@ module ECDSA = struct
     let r, s = Bytes.sub signature 0 32, Bytes.sub signature 32 32 in
     Hacl_ECDSA.hacl_Impl_ECDSA_ecdsa_p256_sha2_verify (C.size_uint32 msg) (C.ctypes_buf msg) (C.ctypes_buf pub) (C.ctypes_buf r) (C.ctypes_buf s)
 end
+
+#ifndef BROKEN_INTRINSICS
+module Chacha20_Poly1305_128 : Chacha20_Poly1305 =
+  Make_Chacha20_Poly1305 (struct
+    let reqs = [AVX]
+    let encrypt = Hacl_Chacha20Poly1305_128.hacl_Chacha20Poly1305_128_aead_encrypt
+    let decrypt = Hacl_Chacha20Poly1305_128.hacl_Chacha20Poly1305_128_aead_decrypt
+  end)
+
+module Chacha20_Poly1305_256 : Chacha20_Poly1305 =
+  Make_Chacha20_Poly1305 (struct
+    let reqs = [AVX2]
+    let encrypt = Hacl_Chacha20Poly1305_256.hacl_Chacha20Poly1305_256_aead_encrypt
+    let decrypt = Hacl_Chacha20Poly1305_256.hacl_Chacha20Poly1305_256_aead_decrypt
+  end)
+
+module Curve25519_64 : Curve25519 =
```

Comment:
can you reuse evercrypt_targetconfig.h for this module and guard the definition of curve64 behind EVERCRYPT_TARGETCONFIG_IS_X64?

Reply:
I gave it a go but the format of evercrypt_targetconfig.h doesn't work with cppo. I made the configure script write IS_X64 to config.h, but this does exclude the `defined(_M_X64)` check.

Reply:
maybe you could write NOT_X64 so that the default remains the same and people can compile on windows without the need to run configure

Reply:
So if i understand correctly this would actually be the same thing as BROKEN_INTRINSICS, which is written in `if ! detect_x64; then`

Reply:
(Actually no sorry that's not exactly true since BROKEN_INTRINISICS is also defined when `! check_no_bug81300`)